controller/src/main/java/org/jboss/as/controller/security/SecurityServiceDescriptor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.jboss.as.controller.security;
+
+import java.security.KeyStore;
+import java.security.Policy;
+
+import javax.net.ssl.SSLContext;
+
+import org.wildfly.security.auth.client.AuthenticationContext;
+import org.wildfly.security.auth.server.HttpAuthenticationFactory;
+import org.wildfly.security.auth.server.SaslAuthenticationFactory;
+import org.wildfly.security.auth.server.SecurityDomain;
+import org.wildfly.security.credential.store.CredentialStore;
+import org.wildfly.service.descriptor.UnaryServiceDescriptor;
+
+/**
+ * Enumerates security related service descriptors.
+ */
+public class SecurityServiceDescriptor {
+
+    public static final UnaryServiceDescriptor<KeyStore> KEY_STORE = UnaryServiceDescriptor.of("org.wildfly.security.key-store", KeyStore.class);
+
+    public static final UnaryServiceDescriptor<Policy> JACC_POLICY = UnaryServiceDescriptor.of("org.wildfly.security.jacc-policy", Policy.class);
```

Comment:
I haven't looked at all the items here, but this one seems off-topic for this PR. I'd want to get an ok from @darranl and @fjuma before moving anything into controller that isn't provided by or used by the server kernel.

Assume the same comment applies to any of these that are for capabilities not provided or used by the kernel.

Scrolling further I realize I want an approval from one of them for this particular commit regardless.

To clarify my general concern: if subsystem X needs a service value of some type found in Java SE, it doesn't care where it comes from, and if it resolves that value using a String capability name it doesn't need to care where it comes from. If X starts relying on the kernel to tell it how to resolve it (via a constant here) now there is a coupling that only exists for the convenience of provided by that constant. The controller module starts to become a kind of registry of capabilities provided by different subsystems.

Reply:
I'll extract this commit to a separate jira and PR so this can be discussed separately.

Reply:
I opened https://issues.redhat.com/browse/WFCORE-6580 to track the security-related service descriptors
and https://issues.redhat.com/browse/WFCORE-6581 to track the jmx-related service descriptors
and dropped these commits from this PR.